contracts/base/AaveV2Wrapper.sol

```diff
@@ -93,79 +109,140 @@ contract AaveV2Wrapper is OwnableUpgradeable, ReentrancyGuardUpgradeable {
+        uint256 decimals = _reserve != ETH
+            ? ICRToken(_reserve).decimals()
+            : ETH_DECIMALS;
+        uint256 aaveLTV = toUSDDecimals(
+            _amount.mul(aaveDefaulLTV).div(BASE),
+            decimals
+        );
+        uint256 usdcAmount = (uint256(assetPrice).mul(aaveLTV).div(1e8));
+        return usdcAmount.mul(borrowRatio).div(BASE);
+    }
+
+    function balanceOf(address _asset) internal view returns (uint256) {
+        if (_asset == ETH) return address(this).balance;
+        else {
+            return IERC20(_asset).balanceOf(address(this));
+        }
     }
 
     //-----------------------------//
     // Internal Mutation Functions //
     //-----------------------------//
 
-    function depositETH(uint256 _amountToAave,uint256 _amountToCruize) internal {
-        wethGateway.depositETH{value: _amountToAave}(address(pool), address(this), 0);
-        payable(cruize_wallet).transfer(_amountToCruize);
+    /**
+     * @dev Deposit the given amount of ETH to the AAVE pool
+     * @param _amount ETH amount
+     */
+    function depositETH(uint256 _amount) internal {
+        wethGateway.depositETH{value: _amount}(address(pool), address(this), 0);
     }
 
-    function depositERC20(address _reserve, uint256 _amountToAave,uint256 _amountToCruize) internal {
-        IERC20 token = IERC20(_reserve);
-        require(token.transferFrom(msg.sender, address(this), _amountToAave));
+    /**
+     * @dev deposit the given amount of asset to the AAVE pool
+     * @param _asset asset address
+     * @param _amountToAave asset amount
+     */
+    function depositERC20(address _asset, uint256 _amountToAave) internal {
+        IERC20 token = IERC20(_asset);
         require(token.approve(address(pool), _amountToAave));
-        pool.deposit(address(token), _amountToAave, address(this), 0);
-        require(token.transferFrom(msg.sender,cruize_wallet, _amountToCruize));
+        pool.deposit(_asset, _amountToAave, address(this), 0);
     }
 
-    function depositToAave(address _reserve, uint256 _amount)
+    /**
+     * @dev deposit given amount erc20 token asset to the AAVE pool
+     * @param _asset asset address
+     * @param _amount asset amount
+     */
+    function depositToAave(address _asset, uint256 _amount)
         internal
         returns (uint256)
     {
         if (_amount == 0) revert ZeroAmount();
-        if (_reserve == address(0)) revert ZeroAddress();
-        if (lpTokens[_reserve] == address(0)) revert AssetNotAllowed();
-        if (msg.value > 0) _amount = msg.value;
-        
-        uint256 amountToCruize = (_amount * 1000) / BASE;
-        uint256 amountToAave = _amount - amountToCruize;
+        if (_asset == address(0)) revert ZeroAddress();
+        if (crTokens[_asset] == address(0)) revert AssetNotAllowed();
 
-        if (_reserve == ETH && msg.value > 0) {
-            depositETH(amountToAave,amountToCruize);
+        if (msg.value > 0) _amount = msg.value;
```

Comment:
i think we don't need this condition here we can use the _amount. = msg.value  in line number 201 because  it work's as same

Reply:
we will not pass msg.value everytime